catroid/src/main/java/org/catrobat/catroid/content/bricks/GoToBrick.java

```diff
@@ -0,0 +1,381 @@
+import android.widget.Spinner;
+import android.widget.SpinnerAdapter;
+import android.widget.TextView;
+
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.BrickValues;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.ui.controller.BackPackSpriteController;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GoToBrick extends BrickBaseType {
+	private static final long serialVersionUID = 1L;
+
+	private Sprite destinationSprite;
+	private transient String oldSelectedObject;
+	private String touchPositionLabel;
+	private String randomPositionLabel;
+
+	private transient AdapterView<?> adapterView;
+	private transient SpinnerAdapterWrapper spinnerAdapterWrapper;
+	private int spinnerSelection;
+
+	public GoToBrick() {
+		this.spinnerSelection = 0;
+		this.oldSelectedObject = "";
+	}
+
+	public GoToBrick(Sprite destinationSprite) {
+		this.destinationSprite = destinationSprite;
+		this.spinnerSelection = 0;
+		this.oldSelectedObject = "";
+	}
+
+	@Override
+	public int getRequiredResources() {
+		return NO_RESOURCES;
+	}
+
+	@Override
+	public Brick copyBrickForSprite(Sprite sprite) {
+		GoToBrick copyBrick = (GoToBrick) clone();
+
+		return copyBrick;
+	}
+
+	@Override
+	public View getView(final Context context, int brickId, BaseAdapter baseAdapter) {
+		if (animationState) {
+			return view;
+		}
+
+		view = View.inflate(context, R.layout.brick_go_to, null);
+		view = getViewWithAlpha(alphaValue);
+
+		this.touchPositionLabel = context.getString(R.string.brick_go_to_touch_position);
+		this.randomPositionLabel = context.getString(R.string.brick_go_to_random_position);
+
+		setCheckboxView(R.id.brick_go_to_checkbox);
+
+		final Brick brickInstance = this;
+		checkbox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+			@Override
+			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+				checked = isChecked;
+				adapter.handleCheck(brickInstance, isChecked);
+			}
+		});
+
+		final Spinner goToSpinner = (Spinner) view.findViewById(R.id.brick_go_to_spinner);
+
+		if (!(checkbox.getVisibility() == View.VISIBLE)) {
+			goToSpinner.setClickable(true);
+			goToSpinner.setEnabled(true);
+		} else {
+			goToSpinner.setClickable(false);
+			goToSpinner.setEnabled(false);
+		}
+
+		final ArrayAdapter<String> spinnerAdapter = createArrayAdapter(context);
+
+		spinnerAdapterWrapper = new SpinnerAdapterWrapper(context, goToSpinner, spinnerAdapter);
+
+		goToSpinner.setAdapter(spinnerAdapterWrapper);
+
+		goToSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+			@Override
+			public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+				String itemSelected = parent.getSelectedItem().toString();
+
+				if (itemSelected.equals(context.getString(R.string.brick_go_to_touch_position))) {
```

Comment:
i don't want to use the labels here, because they're only needed in the setSpinnerSelection method because there's no Context and if i use it everywhere else they're not set in time for some cases